### PR TITLE
Enterprise BYOK

### PR DESCRIPTION
**Pull request by stafot**
#### Summary
Enterprise BYOK

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6817

### CODE REVIEW

File: source/manage/cloud-byok.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+Cloud Dedicated Bring Your Own Key
+===================================
+
+.. include:: ../_static/badges/ent-cloud-dedicated.rst
+   :start-after: :nosearch:
+
+Bring Your Own Key aims to provide dedicated enterprise customers with the capability to introduce their encryption keys and manage their lifecycle within the cloud infrastructure. This approach ensures that customers have full control over their data encryption processes, enhancing data security and compliance. It not only enhances data security but also provides customers with autonomy over their encryption key lifecycle, aligning with the primary goal of supporting encryption at rest with custom KMS keys provided and maintained by the customers. BYOK requires a subscription to Mattermost Cloud Enterprise Dedicated.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:7]:
```suggestion
Bring Your Own Key (BYOK) provides Enterprise Cloud customers with autonomy over their encryption key life cycle. BYOK supports encryption at rest with custom KMS keys that the enterprise provides and maintains.

BYOK requires a subscription to Mattermost Cloud Enterprise Dedicated, which offers enhanced data security and compliance by ensuring that enterprises have full control over their data encryption processes.
```

File: source/manage/cloud-byok.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+In Mattermost Dedicated, you can use KMS keys in two ways:
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:9]:
```suggestion
In Mattermost Cloud Enterprise Dedicated, you can use KMS keys in 2 ways:
```

File: source/manage/cloud-byok.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+- One KMS key for all services
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:11]:
```suggestion
- One KMS key for all services; or,
```

File: source/manage/cloud-byok.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- Per-service KMS keys (EBS, RDS, S3)
+    - Keys do not need to be unique to each service.
+    - All services must be encrypted at rest.
+    - Selective enablement of this feature can be supported.
+    - Keys do not need to be unique to each service.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:16]:
```suggestion
    - Keys don't need to be unique to each service.
```

File: source/manage/cloud-byok.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    - In case a customer needs a global database the recommendation from the Infrastructure SRE team is to provide us 2 KMS keys, 1 per region.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:17]:
```suggestion
    - In cases where a global database is needed, we recommend providing 2 KMS keys (1 per region).
```

File: source/manage/cloud-byok.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Configure BYOK
+------------------------
+1. Customer to provide their AWS KMS ARN to our infrastructure SRE team.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:21]:
```suggestion
1. Enterprise customer provides their AWS KMS ARN to the Mattermost Infrastructure SRE team.
```

File: source/manage/cloud-byok.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+2. Customer need to add the following blocks to their KMS Policy for the AWS KMS ARN provided
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:22]:
```suggestion
2. Enterprise customer adds the following blocks to their KMS Policy for the AWS KMS ARN provided:
```

File: source/manage/cloud-byok.rst [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            "kms:Encrypt",
+            "kms:Decrypt",
+            "kms:ReEncrypt*",
+            "kms:GenerateDataKey*",
+            "kms:DescribeKey"
+        ],
+        "Resource": "<CUSTOM_CUSTOMER_KMS_ID>"
+    },
+
+3. Infrastructure SRE team need to update kops cluster and S3, RDS resources after KMS policy is updated on customer's end.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:57]:
```suggestion
3. The Mattermost Infrastructure SRE team updates the kops cluster and S3, RDS resources after the KMS policy is updated on the customer's end.
```

File: source/manage/cloud-byok.rst [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+An alternative configuration path that Mattermost offers is that the customer provides an external key (non-KMS) to the infrastructure SRE team and the team maintains all the key lifecycle and the relevant resources for the customer.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:59]:
```suggestion
Alternatively, the Enterprise customer can provide an external key (non-KMS) to the Mattermost Infrastructure SRE team that Mattermost maintains on behalf of the customer.
```

File: source/manage/cloud-byok.rst [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+This path offers less control to customers but simplifies the setup process.
+
+Requirements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost)
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:65]:
```suggestion
- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost.)
```

File: source/manage/cloud-byok.rst [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+- Customers oversee the maintenance lifecycle of their custom KMS key.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:66]:
```suggestion
- Customers oversee the maintenance life cycle of their custom KMS key.
```

File: source/manage/cloud-byok.rst [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+- A valid AWS KMS arn for encrypting storage and databases should be provided to the Infrastructure SRE team.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:67]: What is an ``arn``?

File: source/manage/cloud-byok.rst [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+- Incorporate blocks from the Infrastructure SRE team into their KMS key policy.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:68]:
```suggestion
- The customer incorporate blocks from the Infrastructure SRE team into their KMS key policy.
```

File: source/manage/cloud-byok.rst [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+Considerations
+~~~~~~~~~~~~~~~~~~~~~~~
+- Changing the AWS KMS key in the database necessitates downtime due to AWS Aurora's encryption limitations.
+- Proper communication is essential for setting expectations and scheduling changes.
+
+Conclusion
+--------------
+
+If you are a large enterprise with compliance requirements or working in highly-regulated industries using Mattermost Cloud Dedicated with BYOK ensures full data control.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:78]:
```suggestion
If you are a large enterprise with compliance requirements, or are working in highly-regulated industries, using Mattermost Cloud Dedicated with BYOK ensures full data control.
```

**Review by cwarnermm — COMMENTED** [review]
@stafot - I've provided editorial feedback inline. Please review technically to ensure that my proposed edits haven't changed the intended meaning by mistake. Happy to iterate on phrasing in more detail if needed.

File: source/manage/cloud-byok.rst (hunk 12 above)
Comment (stafot) [reply source/manage/cloud-byok.rst:67]: @cwarnermm fyki: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html#

**Review by iyampaul — APPROVED** [review]

File: source/manage/cloud-byok.rst [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+Alternatively, the Enterprise customer can provide an external key (non-KMS) to the Mattermost Infrastructure SRE team that Mattermost maintains on behalf of the customer.
+This path offers less control to customers but simplifies the setup process.
+
+Requirements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost.)
+- Customers oversee the maintenance life cycle of their custom KMS key.
+- A valid AWS KMS arn for encrypting storage and databases should be provided to the Infrastructure SRE team.
```
Comment (cwarnermm) [inline source/manage/cloud-byok.rst:69]:
```suggestion
- A valid AWS KMS ARN for encrypting storage and databases should be provided to the Infrastructure SRE team.
```

**Review by cwarnermm — APPROVED** [review]

File: source/_static/badges/ent-cloud-dedicated.rst [hunk 16]
```diff
@@ -0,0 +1,18 @@
+:orphan:
+:nosearch:
+
+.. raw:: html
+
+  <div class="mm-badge">
+
+|plans-img| Available only on `Enterprise <https://mattermost.com/pricing/>`__ plans
```
Comment (stafot) [inline source/_static/badges/ent-cloud-dedicated.rst:8]: @spirosoik I am not sure if here we want the pricing page or directly the Cloud Enterprise Page.

Comment (stafot) [reply source/_static/badges/ent-cloud-dedicated.rst:8]: @spirosoik Also in this page we might want somowhere the talk to an expert link or button

Comment (spirosoik) [reply source/_static/badges/ent-cloud-dedicated.rst:8]: This needs to redirect you here I believe https://mattermost.com/contact-sales/

Comment (stafot) [reply source/_static/badges/ent-cloud-dedicated.rst:8]: or here https://mattermost.com/enterprise/cloud/ I cannot decide which one fits best

**Review by spirosoik — APPROVED** [review]
few minor comments

File: source/manage/cloud-byok.rst [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+3. The Mattermost Infrastructure SRE team updates the kops cluster and S3, RDS resources after the KMS policy is updated on the customer's end.
+
+Alternatively, the Enterprise customer can provide an external key (non-KMS) to the Mattermost Infrastructure SRE team that Mattermost maintains on behalf of the customer.
+This path offers less control to customers but simplifies the setup process.
+
+Requirements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost.)
```
Comment (stylianosrigas) [inline source/manage/cloud-byok.rst:67]:
```suggestion
- Customers must own their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost.)
```

File: source/manage/cloud-byok.rst [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+BYOK requires a subscription to Mattermost Cloud Enterprise Dedicated, which offers enhanced data security and compliance by ensuring that enterprises have full control over their data encryption processes.
+
+In Mattermost Cloud Enterprise Dedicated, you can use KMS keys in 2 ways:
+
+- One KMS key for all services; or,
+- Per-service KMS keys (EBS, RDS, S3)
+    - Keys do not need to be unique to each service.
+    - All services must be encrypted at rest.
+    - Selective enablement of this feature can be supported.
+    - Keys don't need to be unique to each service.
```
Comment (stylianosrigas) [inline source/manage/cloud-byok.rst:18]:
```suggestion
```

Comment (stylianosrigas) [reply source/manage/cloud-byok.rst:18]: Duplicate

File: source/manage/cloud-byok.rst [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+Alternatively, the Enterprise customer can provide an external key (non-KMS) to the Mattermost Infrastructure SRE team that Mattermost maintains on behalf of the customer.
+This path offers less control to customers but simplifies the setup process.
+
+Requirements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Customers must possess their AWS Account. (In the alternative path mentioned above this is delegated to Mattermost.)
+- Customers oversee the maintenance life cycle of their custom KMS key.
+- A valid AWS KMS ARN for encrypting storage and databases should be provided to the Infrastructure SRE team.
+- The customer incorporate blocks from the Infrastructure SRE team into their KMS key policy.
```
Comment (stylianosrigas) [inline source/manage/cloud-byok.rst:70]:
```suggestion
- The customer should incorporate the provided policy blocks from the Infrastructure SRE team into their KMS key policy.
```

**Review by stylianosrigas — COMMENTED** [review]
Some nit suggestions

**Pull request merged**